### PR TITLE
Adds support for CentOS 8.x in ./packaging/installer/install-required-packages.sh

### LINKED ISSUE #6967: Kickstart.sh doesn't work for CentOS 8 & RHEL 8
##### Bug report summary
Running the `kickstart.sh` on CentOS 8 fails on 'Installation of required packages'.

```
/etc/os-release information:
NAME            : CentOS Linux
VERSION         : 8 (Core)
ID              : centos
ID_LIKE         : rhel fedora
VERSION_ID      : 8

We detected these:
Distribution    : centos
Version         : 8
Codename        : 8 (Core)
Package Manager : install_yum
Packages Tree   : centos
Detection Method: /etc/os-release
Default Python v: 2

This system is receiving updates from Red Hat Satellite or Spacewalk server.
Last metadata expiration check: 2:42:06 ago on Fri Sep 27 14:58:16 2019.
No match for argument: Judy-devel
No match for argument: MySQL-python
No match for argument: autoconf-archive
No match for argument: autogen
No match for argument: libmnl-devel
No match for argument: libuv-devel
No match for argument: python
There are following alternatives for "python": python2, python36
No match for argument: python-psycopg2



We are very sorry!

Installation of required packages failed.

What to do now:

1. Make sure your system is updated.
Most of the times, updating your system will resolve the issue.


2. If the error message is about a specific package, try removing
that package from the command and run it again.
Depending on the broken package, you may be able to continue.


3. Let us know. We may be able to help.
Open a github issue with the above log, at:


https://github.com/firehol/netdata/issues                                                                                             
```
##### OS / Environment
CentOS 8
##### Netdata version (ouput of `netdata -V`)
1.10.0 (I am limited to this version due to company policies. Version should be irrelevant to this issue) 
##### Component Name
Installation scripts
##### Steps To Reproduce
`bash kickstart.sh all --non-interactive`
##### Expected behavior
Installation should succeed.

### PR DESCRIPTION
**Pull request by prologic**
Fixes #6967

**Depends on:** #7916 (_review that one first_)

##### Summary

As per title

##### Component Name

area/ci
area/packaging

##### Additional Information

3/3 PR(s)

##### Test Plan

```sh
$ dki -t --rm -v $PWD:/netdata -w /netdata centos:8 /bin/sh -c './packaging/installer/install-required-packages.sh --dont-wait --non-interactive'
.
.
.
Installed:
  autoconf-2.69-27.el8.noarch                            automake-1.16.1-6.el8.noarch                       gcc-8.3.1-4.5.el8.x86_64
  git-2.18.2-1.el8_1.x86_64                              make-1:4.2.1-9.el8.x86_64                          pkgconf-pkg-config-1.4.2-1.el8.x86_64
  autoconf-archive-2018.03.13-1.el8.noarch               autogen-5.18.12-7.el8.x86_64                       perl-IO-Socket-IP-0.39-5.el8.noarch
  perl-IO-Socket-SSL-2.066-3.el8.noarch                  perl-Mozilla-CA-20160104-7.el8.noarch              autogen-libopts-5.18.12-7.el8.x86_64
  cpp-8.3.1-4.5.el8.x86_64                               gc-7.6.4-3.el8.x86_64                              git-core-2.18.2-1.el8_1.x86_64
  git-core-doc-2.18.2-1.el8_1.noarch                     guile-5:2.0.14-7.el8.x86_64                        isl-0.16.1-6.el8.x86_64
  libatomic_ops-7.6.2-3.el8.x86_64                       libmpc-1.0.2-9.el8.x86_64                          perl-Digest-1.17-395.el8.noarch
  perl-Digest-MD5-2.55-396.el8.x86_64                    perl-Error-1:0.17025-2.el8.noarch                  perl-Git-2.18.2-1.el8_1.noarch
  perl-Net-SSLeay-1.88-1.el8.x86_64                      perl-TermReadKey-2.37-7.el8.x86_64                 perl-Thread-Queue-3.13-1.el8.noarch
  perl-URI-1.73-3.el8.noarch                             perl-libnet-3.11-3.el8.noarch                      emacs-filesystem-1:26.1-5.el8.noarch
  fipscheck-1.5.0-4.el8.x86_64                           fipscheck-lib-1.5.0-4.el8.x86_64                   glibc-devel-2.28-72.el8.x86_64
  glibc-headers-2.28-72.el8.x86_64                       groff-base-1.22.3-18.el8.x86_64                    kernel-headers-4.18.0-147.3.1.el8_1.x86_64
  libedit-3.1-23.20170329cvs.el8.x86_64                  libgomp-8.3.1-4.5.el8.x86_64                       libpkgconf-1.4.2-1.el8.x86_64
  libsecret-0.18.6-1.el8.x86_64                          libtool-ltdl-2.4.6-25.el8.x86_64                   libxcrypt-devel-4.1.1-4.el8.x86_64
  m4-1.4.18-7.el8.x86_64                                 ncurses-6.1-7.20180224.el8.x86_64                  openssh-8.0p1-3.el8.x86_64
  openssh-clients-8.0p1-3.el8.x86_64                     openssl-1:1.1.1c-2.el8.x86_64                      perl-Carp-1.42-396.el8.noarch
  perl-Data-Dumper-2.167-399.el8.x86_64                  perl-Encode-4:2.97-3.el8.x86_64                    perl-Errno-1.28-416.el8.x86_64
  perl-Exporter-5.72-396.el8.noarch                      perl-File-Path-2.15-2.el8.noarch                   perl-File-Temp-0.230.600-1.el8.noarch
  perl-Getopt-Long-1:2.50-4.el8.noarch                   perl-HTTP-Tiny-0.074-1.el8.noarch                  perl-IO-1.38-416.el8.x86_64
  perl-MIME-Base64-3.15-396.el8.x86_64                   perl-PathTools-3.74-1.el8.x86_64                   perl-Pod-Escapes-1:1.07-395.el8.noarch
  perl-Pod-Perldoc-3.28-396.el8.noarch                   perl-Pod-Simple-1:3.35-395.el8.noarch              perl-Pod-Usage-4:1.69-395.el8.noarch
  perl-Scalar-List-Utils-3:1.49-2.el8.x86_64             perl-Socket-4:2.027-3.el8.x86_64                   perl-Storable-1:3.11-3.el8.x86_64
  perl-Term-ANSIColor-4.06-396.el8.noarch                perl-Term-Cap-1.17-395.el8.noarch                  perl-Text-ParseWords-3.30-395.el8.noarch
  perl-Text-Tabs+Wrap-2013.0523-395.el8.noarch           perl-Time-Local-1:1.280-1.el8.noarch               perl-Unicode-Normalize-1.25-396.el8.x86_64
  perl-constant-1.33-396.el8.noarch                      perl-interpreter-4:5.26.3-416.el8.x86_64           perl-libs-4:5.26.3-416.el8.x86_64
  perl-macros-4:5.26.3-416.el8.x86_64                    perl-parent-1:0.237-1.el8.noarch                   perl-podlators-4.11-1.el8.noarch
  perl-threads-1:2.21-2.el8.x86_64                       perl-threads-shared-1.58-2.el8.x86_64              pkgconf-1.4.2-1.el8.x86_64
  pkgconf-m4-1.4.2-1.el8.noarch

Complete!

All Done! - Now proceed to the next step.
```

### CODE REVIEW

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4912961&bid=16262839)

**Comment by prologic** [comment]
Actually given that I've tested the only thing I've changed pretty well here I'm good to merge this. No need to re-test the whole kickstart process IMO.

File: packaging/installer/install-required-packages.sh [hunk 1]
```diff
@@ -1202,6 +1206,61 @@ install_apt_get() {
...
+validate_tree_centos() {
+  local opts=
+  if [ "${NON_INTERACTIVE}" -eq 1 ]; then
+    echo >&2 "Running in non-interactive mode"
+    opts="-y"
+  fi
+
+  echo >&2 " > Checking for epel ..."
+  if ! rpm -qa | grep epel > /dev/null; then
+    if prompt "epel not  found, shall I install it?"; then
```
Comment (knatsakis) [inline packaging/installer/install-required-packages.sh]:
```suggestion
    if prompt "epel not found, shall I install it?"; then
```

File: packaging/installer/install-required-packages.sh [hunk 2]
```diff
@@ -1202,6 +1206,61 @@ install_apt_get() {
 # -----------------------------------------------------------------------------
 # centos / rhel
 
+prompt() {
+  if [ "${NON_INTERACTIVE}" -eq 1 ]; then
+    echo >&2 "Running in non-interactive mode, assuming yes (y)"
+    echo >&2 " > Would have promptedfor ${1} ..."
```
Comment (knatsakis) [inline packaging/installer/install-required-packages.sh]:
```suggestion
    echo >&2 " > Would have prompted for ${1} ..."
```

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4912961&bid=16262839)

File: packaging/installer/install-required-packages.sh [hunk 3]
```diff
@@ -1203,6 +1207,61 @@ install_apt_get() {
...
+    case $yn in
+      [Yy]*) return 0 ;;
+      [Nn]*) return 1 ;;
+      *) echo >&2 "Please answer with yes (y) or no (n)." ;;
+    esac
+  done
+}
+
+validate_tree_centos() {
+  local opts=
```
Comment (knatsakis) [inline packaging/installer/install-required-packages.sh]:
```suggestion
  local opts=""
```

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4912961&bid=16262839)

File: packaging/installer/install-required-packages.sh (hunk 2 above)
Comment (knatsakis) [reply packaging/installer/install-required-packages.sh]: Hmm..for some reason I don't see this applied. The PR diff says that it's not applied

File: packaging/installer/install-required-packages.sh (hunk 3 above)
Comment (knatsakis) [reply packaging/installer/install-required-packages.sh]: Same here. It doesn't seem to have been applied

File: test.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Comment (knatsakis) [inline test.sh]: For some reason test.sh appears again in the PR diff. Please remove.

Comment (prologic) [reply test.sh]: DOne

**Comment by ktsaou** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/netdata/netdata/pullRequest?prid=4912961&bid=16262839)

**Pull request closed without merging**